src/components/InternalAddress.vue

```diff
@@ -0,0 +1,160 @@
+<!--
+  - SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<template>
+	<div>
+		<div v-for="domain in sortedDomains"
+			:key="domain.address">
+			{{ domain.address }}
+			{{ senderType(domain.type) }}
+			<ButtonVue type="tertiary"
+				class="button"
+				:aria-label="t('mail','Remove')"
+				@click="removeInternalAddress(domain)">
+				{{ t('mail','Remove') }}
+			</ButtonVue>
+		</div>
+		<div v-for="email in sortedEmails"
+			:key="email.address">
+			{{ email.address }}
+			{{ t('mail', 'domain') }}
```

Comment:
I'm not a designer but the `user@domain.tld domain [Remove]` line looks a bit wonky.

We could put the `domain` part in parenthesis and make it text max contrast.

Just an idea